api/src/main/java/org/cardanofoundation/rosetta/common/util/CardanoAddressUtils.java

```diff
@@ -169,11 +169,13 @@ public static EraAddressType getEraAddressType(String address) {
     try {
       if (address.startsWith(Constants.ADDRESS_PREFIX)
           || address.startsWith(StakeAddressPrefix.MAIN.getPrefix())) {
+        // validate bech32 address. Unfortunately, it will throw a runtime exception in case of invalid address
+        Bech32.decode(address);
         return EraAddressType.SHELLEY;
       }
       new ByronAddress(address).getAddress();
       return EraAddressType.BYRON;
-    } catch (AddressFormatException | AddressRuntimeException e) {
+    } catch (RuntimeException e) {
```

Comment:
Need to replace with custom ApiException

Reply:
Unfortunately, `Bech32.decode(address)` will throw a runtime exception in case of an invalid address. 
`public static Bech32Data decode(String bech32EncodedString) {
    Tuple<String, byte[]> bech32Data = bech32Decode(bech32EncodedString);
    String hrp = (String)bech32Data._1;
    byte[] b32Arr = (byte[])bech32Data._2;
    if (b32Arr.length < 6) {
      throw new RuntimeException("Invalid data length.");
    } else if (!verifyChecksum(hrp, b32Arr)) {
      throw new RuntimeException("Invalid checksum.");
    } else {
      byte[] data = Arrays.copyOfRange(b32Arr, 0, b32Arr.length - 6);
      byte[] b256Arr = convertBits(data, 5, 8, false);
      if (b256Arr == null) {
        throw new RuntimeException("Invalid data format.");
      } else {
        byte witVer = b32Arr[0];
        return new Bech32Data(hrp, b256Arr, witVer);
      }
    }
  }`
  
  We cannot change it here.